lib/VerticalDelegate.js

```diff
@@ -408,7 +408,8 @@ module.exports = {
 		if (!list.hasReset) return this.reset(list);
 
 		var cpp = this.controlsPerPage(list),
-			end = Math.max(list.$.page1.start, list.$.page2.start) + cpp;
+			end = Math.max(list.$.page1.start, list.$.page2.start) + cpp - 1,
+			idx = props.index - 1;
```

Comment:
Yes. props.index could be previous model count in this scenario. 
But basically, second argument of `modelsAdded()` means newly added model's properties.
Not for existed models. 

Anyway `props.index` indicates the first index which new models are added.
That the reason why we compare it with `end`.
If `props.index` is in current pages, we should update them.
It is line 436.